src/codewhisperer/util/keybindingUtil.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as os from 'os'
+import * as fs from 'fs'
+import * as path from 'path'
+import { getLogger } from '../../shared/logger/logger'
+import * as CodeWhispererConstants from '../models/constants'
+/**
+ * Override keybindings for next, previous suggestion
+ * only when user did not set those keybindings
+ */
+export function overrideKeybindings() {
+    let origin = ''
+    const keybindingsPath = getKeybindingsPath()
+    try {
+        origin = fs.readFileSync(keybindingsPath, 'utf8')
+    } catch (error) {
+        getLogger().error(`Failed to read user keybinding file ${error}`)
+    }
+    try {
+        let overrides = ''
+        const sep = '        '
+        if (!origin.includes(`editor.action.inlineSuggest.showNext`)) {
+            overrides += `${sep}{\n${sep}${sep}"key": "right",\n${sep}${sep}"command": "editor.action.inlineSuggest.showNext",\n${sep}${sep}"when": "inlineSuggestionVisible && !editorReadonly"\n${sep}},\n`
+            overrides += `${sep}{\n${sep}${sep}"key": "alt+]",\n${sep}${sep}"command": "-editor.action.inlineSuggest.showNext",\n${sep}${sep}"when": "inlineSuggestionVisible && !editorReadonly"\n${sep}},\n`
+        }
+        if (!origin.includes(`editor.action.inlineSuggest.showPrevious`)) {
+            overrides += `${sep}{\n${sep}${sep}"key": "left",\n${sep}${sep}"command": "editor.action.inlineSuggest.showPrevious",\n${sep}${sep}"when": "inlineSuggestionVisible && !editorReadonly"\n${sep}},\n`
+            overrides += `${sep}{\n${sep}${sep}"key": "alt+[",\n${sep}${sep}"command": "-editor.action.inlineSuggest.showPrevious",\n${sep}${sep}"when": "inlineSuggestionVisible && !editorReadonly"\n${sep}},\n`
+        }
+
+        if (overrides.length > 0) {
+            let newKeybindings = ''
+            if (origin.includes('[') && origin.includes(']')) {
+                newKeybindings = overrides + origin.substring(origin.indexOf('[') + 1, origin.lastIndexOf(']'))
+            } else {
+                newKeybindings = overrides.slice(0, -2) + '\n'
+            }
+            fs.writeFileSync(keybindingsPath, `[\n${newKeybindings}]`)
+        }
+    } catch (error) {
+        getLogger().error(`Failed to update user keybindings, error ${error}`)
+    }
+}
+
+export function getKeybindingsPath(): string {
+    const home = os.homedir()
+    if (process.platform === 'win32') {
+        return path.join(home, CodeWhispererConstants.keyBindingPathWin)
+    } else if (process.platform === 'darwin') {
+        return path.join(home, CodeWhispererConstants.keyBindingPathMac)
```

Comment:
These paths might be outdated. The VS Code source uses [this](https://github.com/microsoft/vscode/blob/84c46b71a5d292a0a9c6cae60e850088270db37d/src/vs/platform/environment/node/userDataPath.js#L49) and [this](https://github.com/microsoft/vscode/blob/84c46b71a5d292a0a9c6cae60e850088270db37d/src/vs/platform/environment/common/environmentService.ts#L54)

Of course, we should think of alternative solutions at this point. Do you know if adding keybindings via `package.json` + custom `when` clause works? That would be more robust and user-friendly instead of writing to the `keybindings.json` file. We can do that in a separate PR, assuming it works.

Reply:
>Do you know if adding keybindings via package.json + custom when clause works?

It does work. I tested https://github.com/aws/aws-toolkit-vscode/pull/2888/commits/b90b8cf44f2ed708db0cdef1c5902803e552239c